platform_secrets/api.py

```diff
@@ -106,57 +106,65 @@ def _secret_cluster_uri(self) -> str:
     def _get_org_secrets_uri(self, org_name: str) -> str:
         return f"{self._secret_cluster_uri}/{org_name}"
 
-    def _get_user_secrets_uri(self, user: User, org_name: Optional[str]) -> str:
+    def _get_secrets_uri(self, project_name: str, org_name: Optional[str]) -> str:
         if org_name is None:
             base = self._secret_cluster_uri
         else:
             base = self._get_org_secrets_uri(org_name)
-        return f"{base}/{user.name}"
+        return f"{base}/{project_name}"
 
-    def _get_user_secrets_read_perm(
-        self, user: User, org_name: Optional[str]
-    ) -> Permission:
-        return Permission(self._get_user_secrets_uri(user, org_name), "read")
+    def _get_secret_uri(self, secret: Secret) -> str:
+        base = self._get_secrets_uri(secret.project_name, secret.org_name)
+        if secret.owner == secret.project_name:
+            return f"{base}/{secret.key}"
+        return base
+
+    def _get_secret_read_perm(self, secret: Secret) -> Permission:
+        return Permission(self._get_secret_uri(secret), "read")
 
-    def _get_user_secrets_write_perm(
-        self, user: User, org_name: Optional[str]
+    def _get_secrets_write_perm(
+        self, project_name: str, org_name: Optional[str]
     ) -> Permission:
-        return Permission(self._get_user_secrets_uri(user, org_name), "write")
+        return Permission(
+            self._get_secrets_uri(project_name, org_name),
+            "write",
+        )
 
     def _convert_secret_to_payload(self, secret: Secret) -> dict[str, Optional[str]]:
-        return {"key": secret.key, "owner": secret.owner, "org_name": secret.org_name}
+        return {
+            "key": secret.key,
+            "org_name": secret.org_name,
+            "project_name": secret.project_name,
+            # NOTE: We store all user/project keys in one k8s secret.
+            # Project k8s secret can contain keys from multiple users, so
+            # there is no single owner of k8s secret, we loose owner when we work
+            # with project secrets.
+            # For backward compatibility leave `owner` field. It will
+            # contain username for secrets without project.
```

Comment:
this last comment line might not be accurate.

Reply:
What do you mean?

Reply:
I guess I meant that it will be a project name in any case. this is really minor.